### PR TITLE
appveyor: sharness improvements

### PR DESCRIPTION
**Pull request by chriscool**
License: MIT
Signed-off-by: Christian Couder <chriscool@tuxfamily.org>

### CODE REVIEW

**Comment by chriscool** [comment]
I am trying to see what goes wrong and if it might help to disable parallel builds.

**Comment by chriscool** [comment]
It looks like there is a gx related problem:

```
go get -u github.com/whyrusleeping/gx
go get -u github.com/whyrusleeping/gx-go
gx --verbose install --global
make: gx: Command not found
Makefile:22: recipe for target 'deps' failed
make: *** [deps] Error 127
Command exited with code 2
```

**Comment by chriscool** [comment]
There is this error on one of the appveyor builds:

```
[1 / 2] fetched dep: go-net
  - QmWchsfMt9Re1CQaiHqPQC1DrZ9bkpa6n229dRYkGyLXNh not found locally, fetching into c:\gopath\src\gx\ipfs\QmWchsfMt9Re1CQaiHqPQC1DrZ9bkpa6n229dRYkGyLXNh
  - fetching QmWchsfMt9Re1CQaiHqPQC1DrZ9bkpa6n229dRYkGyLXNh via ipfs api
  - fetch finished in 8m3.3306363s
ERROR: [2 / 2 ] parallel fetch: failed to fetch package: QmWchsfMt9Re1CQaiHqPQC1DrZ9bkpa6n229dRYkGyLXNh:open c:\gopath\src\gx\ipfs\QmWchsfMt9Re1CQaiHqPQC1DrZ9bkpa6n229dRYkGyLXNh: The system cannot find the file specified.
ERROR: install deps: failed to fetch dependencies
Makefile:22: recipe for target 'deps' failed
make: *** [deps] Error 1
```

**Comment by chriscool** [comment]
Now the error is the following:

```
expecting success: 
	export IPFS_PATH="$(pwd)/.ipfs" &&
	echo "IPFS_PATH: \"$IPFS_PATH\"" &&
	BITS="2048" &&
	ipfs init --bits="$BITS" >actual_init ||
	test_fsh cat actual_init
 
IPFS_PATH: "/cygdrive/c/go/src/github.com/ipfs/go-ipfs/test/sharness/trash directory.t0020-init.sh/.ipfs"
Error: mkdir /cygdrive/c/go/src/github.com/ipfs/go-ipfs/test/sharness/trash directory.t0020-init.sh/.ipfs: The system cannot find the path specified.
> cat actual_init
initializing ipfs node at /cygdrive/c/go/src/github.com/ipfs/go-ipfs/test/sharness/trash directory.t0020-init.sh/.ipfs
 
not ok 7 - ipfs init succeeds
#	
```

It looks like `ipfs init` doesn't like the IPFS_PATH: "/cygdrive/c/go/src/github.com/ipfs/go-ipfs/test/sharness/trash directory.t0020-init.sh/.ipfs".

**Comment by chriscool** [comment]
This fixes parts of the appveyor problems described in issue #2326.
There are probably a lot more changes needed to make all the test suite pass on Windows, so I think for now we should look at merging this.

**Comment by chriscool** [comment]
@whyrusleeping you should have a look at the Travis CI builds above they all fail on OSX during the `gx --verbose install --global` with:

```
No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.

The build has been terminated
```

This might mean that all those problems on Travis OSX machines are network related.

**Comment by whyrusleeping** [comment]
@chriscool yeah... i think travis and maybe our gateways were having some network isues... theyre green on my latest branch so i'm not sure whats up there

File: test/sharness/t0020-init.sh [hunk 1]
```diff
@@ -19,7 +19,7 @@ test_expect_success "ipfs init fails" '
 	test_must_fail ipfs init 2> init_fail_out
 '
 
-test_expect_success "ipfs init output looks good" '
+test_expect_success POSIX "ipfs init output looks good" '
```
Comment (jbenet) [inline test/sharness/t0020-init.sh]: ooof! we'll need an equivalent test for windows to ensure the lock is taken in a different way

**Comment by jbenet** [comment]
@chriscool one comment above

File: test/sharness/t0020-init.sh (hunk 1 above)
Comment (chriscool) [inline test/sharness/t0020-init.sh]: Yeah, we already check that it fails in the above test.
On Windows the error message is different, so I could test against a different error message.

**Comment by whyrusleeping** [comment]
this LGTM, gonna rebase it on master to make sure the tests are green still after my fixes

**Comment by chriscool** [comment]
@whyrusleeping yeah the tests are green after your rebase

**Comment by chriscool** [comment]
Great job with the fixes!

**Pull request merged**